psychopy/experiment/components/grating/__init__.py

```diff
@@ -160,3 +160,48 @@ def writeInitCode(self, buff):
         depth = -self.getPosInRoutine()
         code += ", depth=%.1f)\n" % depth
         buff.writeIndentedLines(code)
+
+    def writeInitCodeJS(self, buff):
+        # do we need units code?
+        if self.params['units'].val == 'from exp settings':
+            unitsStr = "units : undefined, "
+        else:
+            unitsStr = "units : %(units)s, " % self.params
+
+        # replace variable params with defaults
+        inits = getInitVals(self.params, 'PsychoJS')
+
+        for paramName in inits:
+            val = inits[paramName].val
+            if val is True:
```

Comment:
Is this not handled by `Param`? I feel like it should be - a `bool` param should return in lower case if target is JS, and a `code` param should be run through the transpiler